src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsRulesSpec.groovy

```diff
@@ -869,4 +877,93 @@ class PriceFloorsRulesSpec extends PriceFloorsBaseSpec {
         def bidderRequest = bidder.getBidderRequests(bidRequest.id).last()
         assert bidderRequest.imp[0].bidFloor == 0.04
     }
+
+    def "PBS should populate seatNonBid when bid rejected due to floor"() {
+        given: "PBS config with floors config"
+        def pbsService = pbsServiceFactory.getService(floorsConfig)
+
+        and: "Default BidRequest"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.floors = new ExtPrebidFloors(enforcement: new ExtPrebidPriceFloorEnforcement(enforcePbs: enforcePbs))
+            ext.prebid.returnAllBidStatus = true
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response"
+        def floorValue = PBSUtils.randomFloorValue
+        def floorsResponse = PriceFloorData.priceFloorData.tap {
+            modelGroups[0].values = [(rule): floorValue]
+        }
+        floorsProvider.setResponse(bidRequest.site.publisher.id, floorsResponse)
+
+        and: "PBS cache rules"
+        cacheFloorsProviderRules(pbsService, bidRequest, floorValue)
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid.first().bid[0].price = floorValue - 0.1
+        }
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contain seatNonBid and contain errors"
+        def seatNonBids = response.ext.seatnonbid
+        assert seatNonBids.size() == 1
+
+        def seatNonBid = seatNonBids[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == REJECTED_DUE_TO_PRICE_FLOOR
+        assert seatNonBid.nonBid.size() == bidResponse.seatbid[0].bid.size()
+
+        where:
+        enforcePbs << [true, null]
+    }
+
+    def "PBS shouldn't populate seatNonBid when rejected due to floor and returnAllBidStatus is false"() {
```

Comment:
The case that you propose cannot be parameterized within the same test, since the point of the current test case is bid rejected due to floor. if I set the enforcePbs variable to false, then the floor check will be disabled

Reply:
Sorry, bad wording here. I meant different thing :(